### PR TITLE
[ENG-524] update the --branch option behavior to start collecting lineage for builds

### PR DESCRIPTION
**Pull request by kohsuke**
See the linked design document for more details.

## Release Notes

- **New Features**
	- Enhanced clarity and robustness in session and build management commands.
	- Introduced alias functionality for command groups.
	- Improved handling of command-line options for flavors and links.

- **Bug Fixes**
	- Refined error messages for session name validation and flavor input.
	- Adjusted payload handling for Git submodules and commit options.

- **Tests**
	- Expanded test coverage for build recording and session commands.
	- Added new tests for the `KEY_VALUE` type functionality.

### CODE REVIEW

**Review by ono-max — APPROVED** [review]
Thank you!

File: launchable/commands/record/build.py [hunk 1]
```diff
@@ -237,7 +246,32 @@ def list_submodules(workspaces: List[Workspace]) -> List[Workspace]:
...
+                        err=True)
+                    sys.exit(1)
+
+                if not ws_by_name.get(kv[0]):
+                    click.echo(click.style(
+                        "Invalid repository name {} in a --branch option. ".format(kv[0]),
+                        fg="yellow"),
+                        err=True)
+                    # TODO: is there any reason this is not an error? for now erring on caution
+                    # sys.exit(1)
```
[the remark this answers is not in the document]
Comment (kohsuke) [reply launchable/commands/record/build.py:271]: Nope. Look at the comment.

File: launchable/commands/record/tests.py [hunk 2]
```diff
@@ -166,12 +167,12 @@ def tests(
...
     post_chunk: int,
     subsetting_id: str,
-    flavor,
+    flavor: Sequence[Tuple[str, str]],
     no_base_path_inference: bool,
     report_paths: bool,
     group: str,
     is_allow_test_before_build: bool,
-    links: List[str] = [],
+    links: Sequence[Tuple[str, str]] = [],
```
[the remark this answers is not in the document]
Comment (kohsuke) [reply launchable/commands/record/tests.py:175]: I will incorporate this.

**Comment by kohsuke** [comment]
@ono-max  Thanks for fixing things up. I revisited those a bit. In particular, I believe mixing "" and null is a bad habit, so I fixed that.

**Pull request merged**